livekit-agents/livekit/agents/voice/transcription/_speaking_rate.py

```diff
@@ -0,0 +1,234 @@
+                    },
+                )
+                continue
+
+            if resampler is not None:
+                inference_frames.extend(resampler.push(input_frame))
+            else:
+                inference_frames.append(input_frame)
+
+            while True:
+                available_samples = sum(frame.samples_per_channel for frame in inference_frames)
+                if available_samples < self._window_size_samples:
+                    break
+
+                inference_frame = utils.combine_frames(inference_frames)
+                np.divide(
+                    inference_frame.data[: self._window_size_samples],
+                    np.iinfo(np.int16).max,
+                    out=inference_f32_data,
+                    dtype=np.float32,
+                )
+
+                # run the inference
+                sr = self._compute_speaking_rate(inference_f32_data, _inference_sample_rate)
+                self._event_ch.send_nowait(
+                    SpeakingRateEvent(
+                        timestamp=pub_timestamp,
+                        speaking=sr > 0,
+                        speaking_rate=sr,
+                    )
+                )
+
+                # move the window forward by the hop size
+                pub_timestamp += self._opts.step_size
+                if len(inference_frame.data) - self._step_size_samples > 0:
+                    data = inference_frame.data[self._step_size_samples :]
+                    inference_frames = [
+                        rtc.AudioFrame(
+                            data=data,
+                            sample_rate=inference_frame.sample_rate,
+                            num_channels=1,
+                            samples_per_channel=len(data) // 2,
+                        )
+                    ]
+
+    def _compute_speaking_rate(self, audio: np.ndarray, sample_rate: int) -> float:
+        """
+        Compute the speaking rate of the audio using the selected method
+        """
+        silence_threshold = self._opts._silence_threshold
+
+        audio_sq = audio**2
+        # check if the audio is silent
+        overall_rms = np.sqrt(np.mean(audio_sq))
+        if overall_rms < silence_threshold:
+            return 0.0
+
+        # or if the tail of the audio is silent
+        tail_audio_sq = audio_sq[int(len(audio_sq) * 0.7) :]
+        if len(tail_audio_sq) > 0 and np.sqrt(np.mean(tail_audio_sq)) < silence_threshold * 0.5:
+            return 0.0
+
+        return self._spectral_flux(audio, sample_rate)
+
+    def _spectral_flux(self, audio: np.ndarray, sample_rate: int) -> float:
+        """
+        Calculate speaking rate based on spectral flux.
+        Higher spectral flux correlates with more rapid speech articulation.
+        """
+        # Parameters
+        frame_length = int(sample_rate * 0.025)  # 25ms
+        hop_length = frame_length // 2  # 50% overlap
+
+        # calculate stft
+        _, _, Zxx = signal.stft(
```

Comment:
If we only use scipy for this function. I think this could be easily reproducible with numpy?